src/workers/ans104-data-indexer.ts

```diff
@@ -91,23 +91,53 @@ export class Ans104DataIndexer {
         typeof item.data_size === 'number'
       ) {
         log.debug('Indexing data item data...');
-        await this.contiguousDataIndex.saveDataContentAttributes({
-          id: item.id,
-          hash: item.data_hash,
-          dataSize: item.data_size,
-          contentType: item.content_type,
-        });
-        await this.indexWriter.saveNestedDataId({
-          id: item.id,
-          parentId: item.parent_id,
-          dataOffset: item.data_offset,
-          dataSize: item.data_size,
-        });
-        await this.indexWriter.saveNestedDataHash({
-          hash: item.data_hash,
-          parentId: item.parent_id,
-          dataOffset: item.data_offset,
-        });
+        if (item.data_hash != null) {
+          if (item.data_size != null) {
+            await this.contiguousDataIndex.saveDataContentAttributes({
+              id: item.id,
+              hash: item.data_hash,
+              dataSize: item.data_size,
+              contentType: item.content_type,
+            });
+          } else {
+            log.warn(
+              'Skipping data item data content attributes indexing due to missing data size.',
+            );
+          }
+
+          // Index data hash to parent ID relationship
+          if (item.parent_id != null && item.data_offset != null) {
+            await this.indexWriter.saveNestedDataHash({
+              hash: item.data_hash,
+              parentId: item.parent_id,
+              dataOffset: item.data_offset,
+            });
+          } else {
+            log.warn(
+              'Skipping data item nested data indexing due to missing parent ID or data offset.',
+            );
+          }
+        } else {
+          log.warn('Skipping data item data indexing due to missing hash.');
+        }
+
+        // Index data offset and size for ID to parent ID relationship
+        if (
+          item.parent_id != null &&
+          item.data_offset != null &&
+          item.data_size != null
```

Comment:
nit: since its only one type or the other, I think we could simplify these null checks with a broader `isOptimisiticDataItem` type guard that has a type like `(dataItem: NormalizedDataItem): dataItem is NormalizedOptimisticDataItem => dataItem.parent_id != null`

Reply:
I like the general suggestion, but in this particular case, I'm paranoid about things slipping through with an improper type. I also like that I can look at the list of things being checked here and match them up with the DB calls. That said, I would probably go with the type guard if these conditionals weren't around DB inserts or our I was more confident about our typing overall.